onnxruntime/core/framework/sequential_execution_plan.h

```diff
@@ -35,9 +35,17 @@ struct AllocPlanPerValue {
   AllocPlanPerValue() : location(CPU, OrtArenaAllocator) {}
 };
 
+class MLValueLocator {
+ public:
+  virtual const OrtAllocatorInfo& GetLocation(size_t mlvalue_index) const = 0;
+  // return all memory locations for all the MLValues
+  virtual std::set<OrtAllocatorInfo> GetAllLocations() const = 0;
+  virtual ~MLValueLocator() = default;
+};
+
 // SequentialExecutionPlan: This is the data that is produced by a static
 // planner for a sequential execution, to be used by a SequentialExecutor.
-struct SequentialExecutionPlan {
+struct SequentialExecutionPlan : public MLValueLocator {
```

Comment:
This inheritance relationship doesn't look very intuitive. For e.g. when we see a variable like ```MLValueLocator& execution_plan```. May be it's because of the name. Can we rename it to ExecutionPlanBase?

Reply:
May I change it to 
MLValueLocator& mlvalue_locator.

To me, it's surprising that parrellel executor has a sequential execution plan

Reply:
If I'm reading this correctly it looks like what you want to achieve is to retrieve the location of MLValues without giving full access to the SequentialExecutionPlan (since everything is public in the SequentialExecutionPlan class). However, the inheritance relationship here doesn't model the is-a relationship. One other way to achieve the same thing is to create a wrapper class say PlanProxy that stores an instance of the SequentialExecutionPlan class privately and exposes the same set of methods you want to expose in MLValueLocator. Would this work?